### PR TITLE
Update Bucket and BucketAccess types to latest definitions in the KEP

### PR DESCRIPTION
**Pull request by rrati**

### CODE REVIEW

File: apis/objectstorage.k8s.io/v1alpha1/types.go [hunk 1]
```diff
@@ -274,6 +278,21 @@ type BucketAccessRequestList struct {
 	Items           []BucketAccessRequest `json:"items"`
 }
 
+type ObjectReference struct {
```
Comment (jeffvance) [inline apis/objectstorage.k8s.io/v1alpha1/types.go:281]: Isn't there already a built-in objectreference type?

**Comment by jeffvance** [comment]
lgtm
other than a comment on defining our own objectReference.

File: apis/objectstorage.k8s.io/v1alpha1/types.go (hunk 1 above)
Comment (rrati) [reply apis/objectstorage.k8s.io/v1alpha1/types.go:281]: There is, however defining a different ObjectReference type isn't unheard of.  There are a few in base kube.  That said, this one isn't defining anything that the base ObjectReference doesn't have, and in fact has fewer fields.   I'm indifferent to using the existing vs creating our own.

Comment (jeffvance) [reply apis/objectstorage.k8s.io/v1alpha1/types.go:281]: [objectreference](https://godoc.org/k8s.io/api/core/v1#ObjectReference)
Your type has fewer fields and I'm fine using it is other project are also defining their own.

**Pull request merged**